astropy/units/si.py

```diff
@@ -29,12 +29,22 @@
 ###########################################################################
 # LENGTH
 
+# We exclude c as a prefix so we can define cm as a derived unit,
+# not a PrefixUnit. That way it can be used in cgs as a base unit.
 def_unit(
     ["m", "meter"],
     namespace=_ns,
     prefixes=True,
+    exclude_prefixes=["c"],
     doc="meter: base unit of length in SI",
 )
+def_unit(
+    ["cm", "centimeter"],
```

Comment:
Then this would be

```suggestion
    ["cm"],
```

However that has the complication that `u.cm is u.centimeter`  will fail....

Reply:
Which is probably more important than https://github.com/astropy/astropy/pull/15368#discussion_r1333623877

Reply:
Yeah, I think we want to ensure that `u.cm is u.centimeter`